src/LookingGlassXRWebGLLayer.ts

```diff
@@ -264,43 +302,48 @@ export default class LookingGlassXRWebGLLayer extends XRWebGLLayer {
 			glBindVertexArray(oldVAO)
 		}
 
-		let popup
 		window.addEventListener("unload", () => {
-			if (popup) popup.close()
-			popup = undefined
+			if (cfg.popup) cfg.popup.close()
+			cfg.popup = null
 		})
+
 		const moveCanvasToWindow = (enabled, onbeforeunload) => {
-			if (!!popup == enabled) return
+			if (!!cfg.popup == enabled) return
 
 			if (enabled) {
 				recompileProgram()
-
 				lkgCanvas.style.position = "fixed"
-				lkgCanvas.style.top = "0"
+				lkgCanvas.style.bottom = "0"
 				lkgCanvas.style.left = "0"
-				lkgCanvas.style.width = "100%"
-				lkgCanvas.style.height = "100%"
 
 				lkgCanvas.width = cfg.calibration.screenW.value
 				lkgCanvas.height = cfg.calibration.screenH.value
 
 				document.body.appendChild(controls)
-
-				popup = window.open("", undefined, "width=640,height=360")
-				popup.document.title = "Looking Glass Window (fullscreen me on Looking Glass!)"
-				popup.document.body.style.background = "black"
-				popup.document.body.appendChild(lkgCanvas)
-				console.assert(onbeforeunload)
-				popup.onbeforeunload = onbeforeunload
+				const screenPlacement = "getScreenDetails" in window
```

Comment:
this is windowing logic, consider moving to LookingGlassWindow.ts? 

currently we move the separate functions, `placeWindow`, and `openPopup` to the `LookingGlassWindow.ts` file, however there are still some windowing concerns in the `WebGLLayer` file. 

Ideally, all windowing should be handled in one file for ease of development.

Reply:
this has been addressed here: 

https://github.com/Looking-Glass/looking-glass-webxr/pull/31/commits/7ad0e3230ca4f7a65f045968993b3fbecfe4f845